### PR TITLE
Add endpoint to retrieve quiz results

### PR DESCRIPTION
**Pull request by sofiiadidovych**
# Description

Created the GET endpoint for /api/quiz-results. We can filter data in four ways: 
by fk_answer_id column from the table quiz_results
by fk_user_id column
by both
and without queries - this way we will have all the entries

Fixes # 32

# How to test?

You can use Swagger http://localhost:3000/api/documentation/#/Quiz%20Results/get_quiz_results

# Checklist

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

**Review by TITHI1244 — APPROVED** [review]
The code looks nice and clean. Good job @sofiiadidovych :)

**Review by rajanibusani — COMMENTED** [review]
Hi @sofiiadidovych , I tested Your filter options,all are working nicely...You did  A very good job,👏

**Review by rajanibusani — APPROVED** [review]

**Review by sandhana14 — APPROVED** [review]
Good job @sofiiadidovych 👍

**Comment by sofiiadidovych** [comment]
> The code looks nice and clean. Good job @sofiiadidovych :)

Thank you, @TITHI1244!

**Comment by sofiiadidovych** [comment]
> Hi @sofiiadidovych , I tested Your filter options,all are working nicely...You did A very good job,👏

Thank you, @rajanibusani !

**Comment by sofiiadidovych** [comment]
> Good job @sofiiadidovych 👍

Thank you, @sandhana14 !

File: src/server/api/controllers/quiz-results.controller.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+const knex = require('../../config/db');
+
+const getQuizResults = async (answerId, userId) => {
+  try {
+    let filteredResults;
+    if (answerId !== undefined && userId !== undefined) {
```
Comment (bytewiz) [inline src/server/api/controllers/quiz-results.controller.js:6]: This works fine but you could do this like this as well
```suggestion
    if (answerId && userId) {
```

File: src/server/api/controllers/quiz-results.controller.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      filteredResults = await knex('quiz_results')
+        .where('fk_answer_id', '=', answerId)
+        .where('fk_user_id', '=', userId)
+        .select('id', 'fk_answer_id', 'fk_user_id', 'created_date');
+    } else if (answerId !== undefined) {
```
Comment (bytewiz) [inline src/server/api/controllers/quiz-results.controller.js:11]: same here :)

File: src/server/api/controllers/quiz-results.controller.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      filteredResults = await knex('quiz_results')
+        .where('fk_answer_id', '=', answerId)
+        .select('id', 'fk_answer_id', 'fk_user_id', 'created_date');
+    } else if (userId !== undefined) {
```
Comment (bytewiz) [inline src/server/api/controllers/quiz-results.controller.js:15]: same here :)

**Review by bytewiz — APPROVED** [review]

**Pull request merged**